### ISSUE TITLE
add working sql backing store to docker file

### ISSUE THREAD

**Issue by mjsduncan** [opencog/opencog#1485]
now that the agi-bio project is starting experiments with large atomspaces, it would be great if a working postgresql atomspace database could be installed & configured in the Dockerfile image.  i've been unable to get it to work in the current container through the cogserver or in the unit tests, despite repeatedly following, checking & rechecing the multiple README files.  the test databases exist and are accessable with the users given in the respective config files.  proper permissions are emphasized but not stated explicitly:  what should they be?  the only other clue i've come up with is that for the current build "docker ps" lists the port mapping as "0.0.0.0:17001->17001/tcp; 0.0.0.0:18001->18001/tcp" while /etc/hosts has the usual "127.0.0.1  localhost".  any suggestions?

thanks,
mike

**Comment by linas** [comment]
I think @AmeBel  has this working or mostly working on https://github.com/opencog/docker or at least that was my impression

**Comment by inflector** [comment]
It's not at all obvious to me, having spent a bit of time with it, and
being a docker newb, how to use the dockerfiles in
https://github.com/opencog/docker as compared with the one in opencog/opencog.
The one in opencog/opencog has directions that I can follow and they work.

It would be nice to have a set of instructions for these new docker files.
I'm sure it's obvious to experienced docker users but for me it wasn't at
all obvious how to build using the new files.

**Comment by AmeBel** [comment]
If the instructions @ https://github.com/opencog/docker/blob/master/opencog/README.md are not sufficient enough please let me know.

**Comment by inflector** [comment]
Amen, my issue is that I'm building from a Mac so I didn't know how to
translate the docker-compose as I haven't read up on that. The directions
assume you have some familiarity with docker-compose. I don't even know
that that is at the moment except at a very high level. It seems like some
vagrant-like functionality added to docker for complex installs but since I
barely know docker and my current install works for my current usage, the
additional complexity doesn't seem worth it at the moment.

Directions like:

cd to this directory
type this command

etc.

work great when you are new and don't know how the details work. The
install instructions for installing with docker here:

http://wiki.opencog.org/w/Building_OpenCog

were the first ones I found so I used them and they were easy to follow. I
just installed docker on my Mac and skipped to the section below installing
on Linux and the remaining steps worked fine.

I'm sure it's not all that hard, it's just a new tool, but the other docker
file works so I haven't gone through the trouble of figuring it out yet.
The non-compose Dockerfile worked fine.

I'm running docker in a boot2docker VM which is essentially a
super-lightweight linux VM running under VirtualBox but the docker commands
run on my Mac. For the single Dockerfile, it was easy to figure out.

**Comment by linas** [comment]
for the ROS docker files, I tried to always have 2 or 2 very simple shells scripts: build.sh run.sh and stop.sh  each about 2-10 lines of code.

**Comment by AmeBel** [comment]
The wiki was written before docker-compose based orchestration was added. I just updated it to refer to the README 

According to https://blog.docker.com/2014/10/fig-1-0-boot2docker-compatibility-and-more/ docker-compose( old name fig) should work on mac

docker-comopse should come in handy for users who want to run opencog distributed, as scaling the cogserver instances that would be as simple as `
docker-compose scale=3 cogserver.` . I haven't added  cogserver image to the docker-compose yet though.

**Comment by mjsduncan** [comment]
ok i tried the docker/opencog/README.md instructions and they don't work for me.

1) in "./docker-build.sh" i had to remove "docker build -f" to build the images

2) in "docker-compose.yml" the volume clause makes folders "$OPENCOG_SOURCE_DIR", i had to hard code the paths to make this work.

3) in the instructions for linking to the postgres image it is not clear which environment the commands are to be run in.  the postgres container? the host? the docker container? 

i have a working container built from the main repo opencog/Dockerfile with a postres server running off of the postgres cluster database on the host mounted as a container volume that is accessible from within the container using psql just as described in the persist/sql/README but the test fails with
"Can't perform SQLConnect rc=-1(0) [unixODBC][Driver Manager]Data source name not found, and no default driver specified"
the same error happens when i try using the sample database from the cogserver:  it is accessible with psql but the same error happens.

is there a way to access the host postgres server from within a container running opencog?

**Comment by AmeBel** [comment]
@mjsduncan i have merged changes to the docker repository https://github.com/opencog/docker/pull/33 . The changes include an updated README and odbc.ini entries to make running unit tests simpler once https://github.com/opencog/opencog/pull/1562 is merged.

 In the mean time you can try out the setup by running the steps in https://github.com/opencog/opencog/blob/master/examples/guile/persist-example.scm#L35 (starting from line 35) by replacing  "opencog_test"  with "mycogdata" and "opencog_tester" with "opencog_user", after starting the cogserver
